datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -154,26 +157,26 @@ class except the :py:meth:`check` method but sometimes it might be useful for a
                     self.instances = args[3]
             else:
                 # new-style init: the 3rd argument is `instances`
-                self.instances = args[2]
+                self.instances = list(args[2])
```

Comment:
Because `args` is a `tuple`, but we're storing a `list` above.

Alternatively, since we're not using any of the list-specific operations on `instances` here, we could downgrade the type hint to a `Sequence`…

```python
self.instances = kwargs.get('instances', [])  # type: typing.Sequence[typing.Dict[str, typing.Any]]
```

which would allow us to drop this copy-as-list.